tests/testAutopas/tests/containers/TraversalComparison.cpp

```diff
@@ -0,0 +1,236 @@
+
+  autopasTools::generators::RandomGenerator::fillWithHaloParticles(
+      *container, Molecule({0., 0., 0.}, {0., 0., 0.}, numMolecules), container->getCutoff(), numHaloMolecules);
+
+  container->rebuildNeighborLists(traversal.get());
+
+  if (doSlightShift) {
+    executeSlightShift(container, skin / 2, numMolecules + numHaloMolecules);
+  }
+
+  functor.initTraversal();
+  container->iteratePairwise(traversal.get());
+  functor.endTraversal(newton3Option);
+
+  std::vector<std::array<double, 3>> forces(numMolecules);
+  for (auto it = container->begin(autopas::IteratorBehavior::ownedOnly); it.isValid(); ++it) {
+    EXPECT_TRUE(it->isOwned());
+    forces.at(it->getID()) = it->getF();
+  }
+
+  return {forces, {functor.getUpot(), functor.getVirial()}};
+}
+
+/**
+ * Generates the reference for a simulation configuration that is specified by the given key.
+ * For the reference a linked cells algorithm is used.
+ * @param key The key that specifies the simulation.
+ */
+void TraversalComparison::generateReference(mykey_t key) {
+  auto [numParticles, numHaloParticles, boxMax, doSlightShift] = key;
+  // Calculate reference forces
+  auto [calculatedForces, calculatedGlobals] = calculateForces(
+      autopas::ContainerOption::linkedCells, autopas::TraversalOption::c08, autopas::DataLayoutOption::aos,
+      autopas::Newton3Option::enabled, numParticles, numHaloParticles, boxMax, 1., doSlightShift);
+  _forcesReference[key] = calculatedForces;
+  _globalValuesReference[key] = calculatedGlobals;
+}
+
+/**
+ * This tests a given configuration against a reference configuration.
+ */
+TEST_P(TraversalComparison, traversalTest) {
+  auto [containerOption, traversalOption, dataLayoutOption, newton3Option, numParticles, numHaloParticles, boxMax,
+        cellSizeFactor, doSlightShift] = GetParam();
+
+  // empirically determined and set near the minimal possible value for 2000 particles
+  // i.e. if something changes, it may be needed to increase value
+  // (and OK to do so)
+  double rel_err_tolerance = 1.0e-11;
+  double rel_err_tolerance_globals = 1.0e-12;
+
+  auto [calculatedForces, calculatedGlobals] =
+      calculateForces(containerOption, traversalOption, dataLayoutOption, newton3Option, numParticles, numHaloParticles,
+                      boxMax, cellSizeFactor, doSlightShift);
+  if (calculatedForces.empty()) {
+    GTEST_SKIP_("Not applicable!");
+  }
+
+  TraversalComparison::mykey_t key{numParticles, numHaloParticles, boxMax, doSlightShift};
+  if (_forcesReference.count(key) == 0) {
+    generateReference(key);
+  }
+
+  for (size_t i = 0; i < numParticles; ++i) {
+    for (unsigned int d = 0; d < 3; ++d) {
+      double calculatedForce = calculatedForces[i][d];
+      double referenceForce = _forcesReference[key][i][d];
+      EXPECT_NEAR(calculatedForce, referenceForce, std::fabs(calculatedForce * rel_err_tolerance))
+          << "Particle id: " << i;
```

Comment:
why not print the whole particle

Reply:
because I don't have the entire particle here.